### ISSUE TITLE
Expand check BC_GCP_GENERAL_3 - GKE cluster should have shielded nodes 

### ISSUE THREAD

**Issue by Raclaw-jl** [bridgecrewio/checkov#618]
**Describe the solution you'd like**
There is no check to see if shielded nodes are enabled for a GKE cluster at the moment. 

**Describe alternatives you've considered**
Create a new check specifically for instances used by a GKE cluster under the Kubernetes section.

**Additional context**
Terraform code used:
```go
resource "google_container_cluster" "test-cluster" {
  #checkov:skip=CKV_GCP_19: False positive -- BROKEN CHECK
  #checkov:skip=CKV_GCP_13: Not needed, bearer tokens will be disbaled in 1.18 versions
  #checkov:skip=CKV_GCP_20: Not needed
  #checkov:skip=CKV_GCP_23: Not needed
  #checkov:skip=CKV_GCP_25: Not needed

  name     = "test-cluster"
  location = "europe-west1-c"
  provider = google-beta

  remove_default_node_pool = true
  initial_node_count       = 1


  # ip_allocation_policy ## Not needed

  resource_labels = {
    env           = "prod"
    instance-type = "preemptible"
  }

  release_channel {
    channel = "RAPID"
  }

  pod_security_policy_config {
    enabled = true
  }

  // private_cluster_config {
  //   enable_private_nodes    = true
  //   enable_private_endpoint = true
  // }

  master_auth {
    username = ""
    password = ""

    client_certificate_config {
      issue_client_certificate = false
    }
  }
}
```
As you can see, the shielded instances check has not been triggered.
```bash
➜  ~  checkov -d path/to/terraform/code

       _               _              
   ___| |__   ___  ___| | _______   __
  / __| '_ \ / _ \/ __| |/ / _ \ \ / /
 | (__| | | |  __/ (__|   < (_) \ V / 
  \___|_| |_|\___|\___|_|\_\___/ \_/  
                                      
By bridgecrew.io | version: 1.0.589 

terraform scan results:

Passed checks: 10, Failed checks: 0, Skipped checks: 5

Check: CKV_GCP_24: "Ensure PodSecurityPolicy controller is enabled on the Kubernetes Engine Clusters"
        PASSED for resource: google_container_cluster.primaris
        File: /cluster.tf:1-43

Check: CKV_GCP_18: "Ensure GKE Control Plane is not public"
        PASSED for resource: google_container_cluster.primaris
        File: /cluster.tf:1-43

Check: CKV_GCP_8: "Ensure Stackdriver Monitoring is set to Enabled on Kubernetes Engine Clusters"
        PASSED for resource: google_container_cluster.primaris
        File: /cluster.tf:1-43

Check: CKV_GCP_1: "Ensure Stackdriver Logging is set to Enabled on Kubernetes Engine Clusters"
        PASSED for resource: google_container_cluster.primaris
        File: /cluster.tf:1-43

Check: CKV_GCP_21: "Ensure Kubernetes Clusters are configured with Labels"
        PASSED for resource: google_container_cluster.primaris
        File: /cluster.tf:1-43

Check: CKV_GCP_7: "Ensure Legacy Authorization is set to Disabled on Kubernetes Engine Clusters"
        PASSED for resource: google_container_cluster.primaris
        File: /cluster.tf:1-43

Check: CKV_GCP_12: "Ensure Network Policy is enabled on Kubernetes Engine Clusters"
        PASSED for resource: google_container_cluster.primaris
        File: /cluster.tf:1-43

Check: CKV_GCP_10: "Ensure 'Automatic node upgrade' is enabled for Kubernetes Clusters"
        PASSED for resource: google_container_node_pool.primaris_preemptible_nodes
        File: /cluster.tf:45-80

Check: CKV_GCP_22: "Ensure Container-Optimized OS (cos) is used for Kubernetes Engine Clusters Node image"
        PASSED for resource: google_container_node_pool.primaris_preemptible_nodes
        File: /cluster.tf:45-80

Check: CKV_GCP_9: "Ensure 'Automatic node repair' is enabled for Kubernetes Clusters"
        PASSED for resource: google_container_node_pool.primaris_preemptible_nodes
        File: /cluster.tf:45-80

Check: CKV_GCP_25: "Ensure Kubernetes Cluster is created with Private cluster enabled"
        SKIPPED for resource: google_container_cluster.primaris
        Suppress comment:  Not needed
        File: /cluster.tf:1-43

Check: CKV_GCP_23: "Ensure Kubernetes Cluster is created with Alias IP ranges enabled"
        SKIPPED for resource: google_container_cluster.primaris
        Suppress comment:  Not needed
        File: /cluster.tf:1-43

Check: CKV_GCP_20: "Ensure master authorized networks is set to enabled in GKE clusters"
        SKIPPED for resource: google_container_cluster.primaris
        Suppress comment:  Not needed
        File: /cluster.tf:1-43

Check: CKV_GCP_13: "Ensure a client certificate is used by clients to authenticate to Kubernetes Engine Clusters"
        SKIPPED for resource: google_container_cluster.primaris
        Suppress comment:  Not needed, bearer tokens will be disbaled in 1.18 versions
        File: /cluster.tf:1-43

Check: CKV_GCP_19: "Ensure GKE basic auth is disabled"
        SKIPPED for resource: google_container_cluster.primaris
        Suppress comment:  False positive -- BROKEN CHECK
        File: /cluster.tf:1-43
```
